### PR TITLE
Add option to allow only partial use to the `allowTop` option of `regexp/no-useless-non-capturing-group` rule.

### PR DESCRIPTION
**Pull request by ota-meshi**
Related to https://github.com/ota-meshi/eslint-plugin-regexp/issues/181#issuecomment-838271791.

### CODE REVIEW

File: lib/rules/no-useless-non-capturing-group.ts [hunk 1]
```diff
@@ -56,10 +70,29 @@ export default createRule("no-useless-non-capturing-group", {
             node,
             getRegexpLocation,
             fixReplaceNode,
+            getUsageOfPattern,
         }: RegExpContext): RegExpVisitor.Handlers {
+            let isIgnore: (gNode: Group) => boolean
```
Comment (RunDevelopment) [inline lib/rules/no-useless-non-capturing-group.ts:75]: Pedantic nit...

```suggestion
            let isIgnored: (gNode: Group) => boolean
```

File: lib/rules/no-useless-non-capturing-group.ts [hunk 2]
```diff
@@ -56,10 +70,29 @@ export default createRule("no-useless-non-capturing-group", {
...
+            if (allowTop === "always") {
+                isIgnore = isTopLevel
+            } else if (allowTop === "partial") {
+                const usageOfPattern = getUsageOfPattern()
+                if (
+                    usageOfPattern === UsageOfPattern.partial ||
+                    usageOfPattern === UsageOfPattern.mixed
+                ) {
+                    isIgnore = isTopLevel
```
Comment (RunDevelopment) [inline lib/rules/no-useless-non-capturing-group.ts:84]: I would prefer this to be more conservative. The usage might be `UsageOfPattern.unknown`. If we don't know the usage, then we shouldn't assume that it's okay to remove the top-level group.

```suggestion
                if (usageOfPattern !== UsageOfPattern.whole) {
                    isIgnore = isTopLevel
```

**Review by RunDevelopment — CHANGES_REQUESTED** [review]
Thank you @ota-meshi!

Two minor things but apart from that good to go.

**Comment by ota-meshi** [comment]
Thank you for your review! I updated it, so could you check it again?

**Comment by ota-meshi** [comment]
I noticed that I haven't changed about autofix yet. I will fix it again.

**Comment by ota-meshi** [comment]
>I noticed that I haven't changed about autofix yet. I will fix it again.

I fixed it.

**Comment by RunDevelopment** [comment]
Thank you @ota-meshi!

**Pull request merged**